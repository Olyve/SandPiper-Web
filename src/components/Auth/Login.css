.login {
  display: flex;
  flex-direction: column;
  margin: 10px;
  background-color: #ccc;
  border-radius: 10px;
  width: 300px;
  height: 175px;
}

form {
  flex: 4;
  display: flex;
  flex-direction: column;
}

form div {
  flex: 1;
  height: 20px;
}

h2 {
  flex: 1;
  color: #222;
}