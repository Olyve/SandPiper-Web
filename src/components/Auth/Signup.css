.signup {
  margin: 10px;
}

.submit {
  display: inline-block;
}

form {
  vertical-align: center;
}

label input {
  margin-bottom: 10px;
  margin-left: 10px;
}