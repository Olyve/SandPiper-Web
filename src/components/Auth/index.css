.auth {
  display: flex;
  flex-direction: column;
  height: 100%;
  align-items: center;
  justify-content: center;
}

h4 a {
  color: blue;
  text-decoration: underline;
}

h4 a:hover {
  background-color: #bbb;
  color: white;
}