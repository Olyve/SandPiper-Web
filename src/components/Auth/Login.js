import React, { Component } from 'react';
import './Login.css';

class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: ''
    };

    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  onChange(event) {
    let target = event.target;
    let value = target.value;
    let name = target.name;

    this.setState({
      [name]: value
    });
  }

  handleSubmit(event) {
    this.props.handleLogin({
      email: this.state.email,
      password: this.state.password
    });
    this.setState({
      email: '',
      password: ''
    });

    event.preventDefault();
  }

  render() {
    return (
      <div className='login'>
        <h2>Login</h2>
        <form>
          <div>
            <label>Email: </label>
            <input type='text' name='email' value={this.state.email} onChange={this.onChange} />
          </div>
          <div>
            <label>Password: </label>
            <input type='password' name='password' value={this.state.password} onChange={this.onChange} />
          </div>
          <div>
            <button className='submit' onClick={this.handleSubmit}>Submit</button>
          </div>
        </form>
      </div>
    );
  }
}

export default Login;