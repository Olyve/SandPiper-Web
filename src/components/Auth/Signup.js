import React, { Component } from 'react';
import './Signup.css';

class Signup extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: ''
    };

    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  onChange(event) {
    let target = event.target;
    let value = target.value;
    let name = target.name;

    this.setState({
      [name]: value
    });
  }

  handleSubmit(event) {
    this.props.handleSignup({ ...this.state });

    event.preventDefault();
  }

  render() {
    return (
      <div className='signup'>
        <form>
          <label>Email: 
          <input type='text' name='email' onChange={this.onChange}/>
          </label>
          <br />
          <label>Password: 
          <input type='password' name='password' onChange={this.onChange}/>
          </label>
          <br />
          <button className='submit' onClick={this.handleSubmit}>Submit</button>
        </form>
      </div>
    );
  }
}

export default Signup;