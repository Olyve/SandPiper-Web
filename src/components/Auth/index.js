import React, {Component} from 'react';
import Signup from './Signup';
import Login from './Login';
import './index.css';


class Auth extends Component {
  constructor(props) {
    super(props);

    this.state = {
      showLogin: true
    };

    this.toggleView = this.toggleView.bind(this);
  }

  toggleView() {
    this.setState({showLogin: !this.state.showLogin});
  }

  render() {
    var component;
    var message;

    if (this.state.showLogin) {
      component = <Login handleLogin={this.props.handleLogin} />;
      message = <h4>If you don't have an account <a onClick={this.toggleView}>click here</a> to register.</h4>;
    }
    else {
      component = <Signup handleSignup={this.props.handleSignup} />;
      message = <h4>If you already have an account <a onClick={this.toggleView}>click here</a> to log in.</h4>;
    }

    return (
      <div className='auth'>
        {component}
        {message}
      </div>
    );
  }
}


export default Auth;