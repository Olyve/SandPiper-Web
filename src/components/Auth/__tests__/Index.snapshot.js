import React from 'react';
import renderer from 'react-test-renderer';
import Auth from '../index';

describe('Auth', () => {
  it('should render correctly', () => {
    const rendered = renderer.create(
      <Auth />
    );

    expect(rendered.toJSON()).toMatchSnapshot();
  });
});