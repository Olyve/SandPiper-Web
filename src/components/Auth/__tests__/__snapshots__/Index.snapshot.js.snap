// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth should render correctly 1`] = `
<div
  className="auth"
>
  <div
    className="login"
  >
    <h2>
      Login
    </h2>
    <form>
      <label>
        Email:
        <input
          name="email"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <br />
      <label>
        Password:
        <input
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
      </label>
      <br />
      <button
        className="submit"
        onClick={[Function]}
      >
        Submit
      </button>
    </form>
  </div>
  <div
    className="signup"
  >
    <h2>
      Register New User
    </h2>
    <form>
      <label>
        Email:
        <input
          name="email"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <br />
      <label>
        Password:
        <input
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
      </label>
      <br />
      <button
        className="submit"
        onClick={[Function]}
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
